### PR TITLE
Fetch: redirects to data: URLs

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/fetch/issues/393.

### CODE REVIEW

File: fetch/redirects/data.window.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// See ../api/redirect/redirect-to-dataurl.any.js for fetch() tests
+
+async_test(t => {
+  const img = document.createElement("img");
+  img.onload = t.unreached_func();
+  img.onerror = t.step_func_done();
+  img.src = "../api/resources/redirect.py?location=data:image/png%3Bbase64,iVBORw0KGgoAAAANSUhEUgAAAIUAAABqCAIAAAAdqgU8AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAF6SURBVHhe7dNBDQAADIPA%2Bje92eBxSQUQSLedlQzo0TLQonFWPVoGWjT%2BoUfLQIvGP/RoGWjR%2BIceLQMtGv/Qo2WgReMferQMtGj8Q4%2BWgRaNf%2BjRMtCi8Q89WgZaNP6hR8tAi8Y/9GgZaNH4hx4tAy0a/9CjZaBF4x96tAy0aPxDj5aBFo1/6NEy0KLxDz1aBlo0/qFHy0CLxj/0aBlo0fiHHi0DLRr/0KNloEXjH3q0DLRo/EOPloEWjX/o0TLQovEPPVoGWjT%2BoUfLQIvGP/RoGWjR%2BIceLQMtGv/Qo2WgReMferQMtGj8Q4%2BWgRaNf%2BjRMtCi8Q89WgZaNP6hR8tAi8Y/9GgZaNH4hx4tAy0a/9CjZaBF4x96tAy0aPxDj5aBFo1/6NEy0KLxDz1aBlo0/qFHy0CLxj/0aBlo0fiHHi0DLRr/0KNloEXjH3q0DLRo/EOPloEWjX/o0TLQovEPPVoGWjT%2BoUfLQIvGP/RoGWjR%2BIceJQMPIOzeGc0PIDEAAAAASUVORK5CYII";
+}, "<img> fetch that redirects to data: URL");
+
+globalThis.globalTest = null;
```
Comment (yutakahirano) [inline fetch/redirects/data.window.js:10]: How about 

globalThis.testObjects = {};

...

const id = token();
globalThis.testObjects[id] = t;
script.src = \`...,(globalThis.testObjects[${id)].unreached_func())()\`;

?

Comment (annevk) [reply fetch/redirects/data.window.js:10]: That doesn't seem needed until more tests that require this are added?

Comment (yutakahirano) [reply fetch/redirects/data.window.js:10]: I want tests to be copy-pastable, when possible.

Comment (annevk) [reply fetch/redirects/data.window.js:10]: Wouldn't this require that you also copy the import for `token()` and such? It seems that would make it less copy-and-pastable.

Comment (yutakahirano) [reply fetch/redirects/data.window.js:10]: OK, then this LGTM.

**Review by yutakahirano — APPROVED** [review]

**Pull request merged**